src/app/pages/finish/containers/finish-page.component.ts

```diff
@@ -57,37 +57,45 @@ export class FinishPageComponent {
 
   processDataAndSend() {
     this.finishTaskService.updateTaskToComplete(this.associatedTask)
-    if (!this.associatedTask.name.includes('DEMO'))
-      return this.prepareDataService
-        .processQuestionnaireData(
-          this.questionnaireData.answers,
-          this.questionnaireData.timestamps
-        )
-        .then(data =>
-          this.sendToKafka(
-            this.associatedTask,
-            data,
-            this.questionnaireData.questions
-          )
-        )
+    if (!this.associatedTask.name.includes('DEMO')) {
+      const data = this.prepareDataService.processQuestionnaireData(
+        this.questionnaireData.answers,
+        this.questionnaireData.timestamps
+      )
+      this.firebaseAnalytics.logEvent('processed_questionnaire_data', {
```

Comment:
Chain this with then() perhaps ?

Reply:
'processQuestionnaireData()' does not return a promise anymore which is why the chaining has been removed.

Reply:
ok thanks